StreamChatClient_v3/Controllers/ChannelListControllerPublishers.swift

```diff
@@ -0,0 +1,121 @@
+        let controller = ChannelListControllerGeneric(query: query, client: self)
+        return .init(controller: controller)
+    }
+}
+
+/// Describes possible states of Controller's remote activity.
+public enum RemoteActivity {
+    /// The controller is not listening for updates. Call `startUpdating` to fetch the remote data and start listening for changes.
+    case none
+    
+    /// Controller actively tries to refresh the local data.
+    case fetchingRemoteData
+    
+    /// Controller is listening for incoming changes.
+    case listening
+    
+    /// Remote data fetch failed. Controller might still be able to show the recent incoming changes, but it's possible
+    /// the presented data are not complete.
+    case failed(error: Error?)
+}
+
+@available(iOS 13, *)
+extension ChannelListControllerGeneric {
+    /// An object which wraps `ChannelListController` and exposes its properties as Publishers.
+    public class Publishers {
+        /// Publishes changes to the remote activity state. You can use this publisher to show activity indicator to users
+        /// if the controller is fetching the remote data.
+        public var remoteActivityPublisher: AnyPublisher<RemoteActivity, Never> {
+            remoteActivity
+                // Is something like this the way to keep `Publishers` object alive if there is a subscriber?
+                .map { [self] in _ = self; return $0 }
+                .eraseToAnyPublisher()
+        }
+        
+        /// Publishes the changes to the list of channels matching the query. You can use this publisher to animate changes
+        /// to the channel list, or to simply reload all your the data.
+        ///
+        /// Alternatively, you can use `channelsDiffableDatasourcePublisher` which provides similar functionality using the
+        /// `NSDiffableDataSource` API.
+        ///
+        /// - Note: ⚠️ The publisher publishes only **changes** to the list of channels. Use the `channels` property to load
+        /// the initial data before subscribing to the publisher.
+        public var channelChangesPublisher: AnyPublisher<[Change<ChannelModel<ExtraData>>], Never> {
+            channelChanges
+                .map { [self] in _ = self; return $0 }
+                .eraseToAnyPublisher()
+        }
+        
+        // TODO: Implement
+        public var channelsDiffableDatasourcePublisher:
+            AnyPublisher<NSDiffableDataSourceSnapshot<ChannelListSection, ChannelModel<ExtraData>>, Never>! = nil
+        
+        /// The channels matching the query. If you want to react to changes in this field, subscribe to `channelChanges` publisher.
+        public var channels: [ChannelModel<ExtraData>] {
+            // Start observing changes if needed
+            if controller.state == .idle {
+                controller.startUpdating()
+            }
+            
+            return controller.channels
+        }
```

Comment:
I don't this this should be here, because it's replicate `ChannelListController.channels`

Reply:
Ok, I see a developer can create publisher from the Client, but what is the value to have it here if he decide to use Combine?